### DISCUSSION TITLE
Problema Lab 4.02 task 1

### DISCUSSION THREAD

**Discussion by chape8** [rramosp/ai4eng.v1#206] [04 - GESTIÓN DE DATOS]
Estoy un poco atascado con la solución de la primera tarea de este lab, ya que me sale que resultado incorrecto a la hora de la calificación. Pruebo con el ejemplo que colocan ahí del edificio 900, la suma de la variable objetivo me da lo que indica el ejemplo que es 4941, pero al sumar el resto me da un valor de 552190.2, valor inferior al que se supone que debe dar, y la verdad no se cual es el error que estoy cometiendo al momento de quizá sumar o filtrar el resto de variables
![t1](https://user-images.githubusercontent.com/98787315/197361876-f43792da-4544-4def-8fd2-825e73311b20.png)

Este es el filtro para el weather_train
![weather_train](https://user-images.githubusercontent.com/98787315/197361948-38c50a90-ff7f-4263-baae-ff691231012f.png)

Este es el que realizo para el build_metadata
![metadata](https://user-images.githubusercontent.com/98787315/197361988-666862e5-b812-49fc-9cce-a30e1f6617d8.png)

Y este es el que realicé para el train
![train](https://user-images.githubusercontent.com/98787315/197362001-f2517a70-a854-4275-818e-0cfac4e6d659.png)

**Comment by imEag** [comment]
Hola! Para responde quisiera ver cual es la forma resultante de tu dataframe final, por favor adjunta una imagen.

**Reply by chape8** [reply]
![df_result](https://user-images.githubusercontent.com/98787315/197844939-41da19d3-7945-4321-b714-15fa7ccdc568.png)

**Comment by imEag** [comment]
Hola! También necesito que me muestre como obtuviste las sumas, es decir, las variables `target` y `X`. 

Además te recomiendo que reemplaces el índice de `df_result` por la fecha y la hora, esto lo puedes hacer creando un date range con `date_range = pd.date_range(start=date, freq='1h', periods=24)` donde `date` es la fecha que se te asigna en la tarea y luego reemplazando el indice original con el método [set_index](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.set_index.html) ( aunque esto no debería afectar el resultado de las sumas).

**Reply by chape8** [reply]
Aquí está como obtuve las sumas de las variables target y X

![target_X](https://user-images.githubusercontent.com/98787315/198165883-1d09f8a6-cc64-4939-b6dd-cdea69e1207e.png)

**Comment by simonpatino** [comment]
Tu problema esta en que estas repitiendo elementos de la columna timestamp o otra manera de verlo es que estas utilizando todos los site_id en df_result. Solo debes usar los site_id que le corresponden a tu  building_id.
Puedes revisar con:
`meta = pd.read_csv("building_metadata.csv")`
`meta[meta.building_id == "Tu building id"] `

**Reply by imEag** [reply]
Intenta con esto y nos cuentas como te va @chape8

**Reply by imEag** [reply]
una vez hallas seleccionado el site_id que te toca, muéstrame de nuevo como se ve el dataset resultante

**Reply by chape8** [reply]
En el build_metadata me indicaba que era el site_id=9
![build_metadata](https://user-images.githubusercontent.com/98787315/198346011-bb9c93f9-34da-4e33-a7cb-f204f8250999.png)

Ya aquí hice el filtro para el site_id
![filtrar site id](https://user-images.githubusercontent.com/98787315/198345735-f99ca560-d5af-4401-8444-637a3f029415.png)

Y aquí está el df_result donde ya tiene 24 filas correspondientes a cada hora del dia seleccionado( que es el del ejemplo)
![df_result](https://user-images.githubusercontent.com/98787315/198346526-8d5efe60-6781-4176-9407-ddd61a567499.png)

Y aquí el valor del target y X, se sigue manteniendo igual el valor del target y ya el valor de X se redujo( creo que era esperado pues se eliminaron los demás site_id que no interesaban)
![target_X](https://user-images.githubusercontent.com/98787315/198346837-765d0313-7d22-4d4c-a0d6-f3c93d6acd30.png)

**Reply by imEag** [reply]
vale, perfecto, ahora, lo que debes hacer es tomar el `square_feet` de la primera fila, multiplicarlo por 23 y sumarlo al resultado final

**Reply by imEag** [reply]
esto es porque este valor debe estar en cada fila pero solo esta en la primera

**Reply by chape8** [reply]
Ya me dio el resultado, muchísimas gracias por la ayuda.
imEag me podrías hacer el favor de  colaborarme con otra duda que tengo en el Lab3.01 task 2, yo ya hice la consulta en el foro, me habían empezado a colaborar pero ya después no me siguieron colaborando
